rasa/utils/tensorflow/model_data_utils.py

```diff
@@ -1,226 +1,370 @@
+                for list_of_list_of_features in all_features
+                for list_of_features in list_of_list_of_features
                 if list_of_features is not None
             ]
         )
     )
 
-    # create zero_features for nones
-    zero_features = []
-    for features in example_features:
-        new_features = copy.deepcopy(features)
-        if features.is_dense():
-            new_features.features = np.zeros_like(features.features)
-        if features.is_sparse():
+    # create fake_features for Nones
+    fake_features = []
+    for _features in example_features:
+        new_features = copy.deepcopy(_features)
+        if _features.is_dense():
+            new_features.features = np.zeros(
+                (0, _features.features.shape[-1]), _features.features.dtype
+            )
+        if _features.is_sparse():
             new_features.features = scipy.sparse.coo_matrix(
-                features.features.shape, features.features.dtype
+                (0, _features.features.shape[-1]), _features.features.dtype
             )
-        zero_features.append(new_features)
+        fake_features.append(new_features)
 
-    return zero_features
+    return fake_features
 
 
 def convert_to_data_format(
-    tracker_state_features: Union[
+    features: Union[
         List[List[Dict[Text, List["Features"]]]], List[Dict[Text, List["Features"]]]
     ],
-    zero_state_features: Optional[Dict[Text, List["Features"]]] = None,
+    fake_features: Optional[Dict[Text, List["Features"]]] = None,
+    consider_dialogue_dimension: bool = True,
+    featurizers: Optional[List[Text]] = None,
 ) -> Tuple[Data, Optional[Dict[Text, List["Features"]]]]:
     """Converts the input into "Data" format.
 
+    "features_for_examples" can, for example, be a dictionary of attributes (INTENT,
+    TEXT, ACTION_NAME, ACTION_TEXT, ENTITIES, SLOTS, FORM) to a list of features for
+    all dialogue turns in all training trackers.
+    For NLU training it would just be a dictionary of attributes (either INTENT or
+    RESPONSE, TEXT, and potentially ENTITIES) to a list of features for all training
+    examples.
+
     Args:
-        tracker_state_features: a dictionary of attributes (INTENT, TEXT, ACTION_NAME,
-            ACTION_TEXT, ENTITIES, SLOTS, FORM) to a list of features for all
-            dialogue turns in all training trackers
-        zero_state_features: Contains default feature values for attributes
+        features: a dictionary of attributes to a list of features for all
+            examples in the training data
+        fake_features: Contains default feature values for attributes
+        consider_dialogue_dimension: If set to false the dialogue dimension will be
+            removed from the resulting sequence features.
+        featurizers: the featurizers to consider
 
     Returns:
-        Input in "Data" format and zero state features
+        Input in "Data" format and zero features

```

Comment:
What are zero features? Do you mean fake features, a.k.a. default features?

Reply:
yeah, forgot to change the comment

Reply:
@Ghostvv could you please take this over

Reply:
renamed to fake